src/Persistence/PaYaml/Serialization/PFxExpressionYamlConverter.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+using YamlDotNet.Core.Events;
+using Microsoft.PowerPlatform.PowerApps.Persistence.PaYaml.Models.PowerFx;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.PaYaml.Serialization;
+
+// BUG 27469059: Internal classes not accessible to test project. InternalsVisibleTo attribute added to csproj doesn't get emitted because GenerateAssemblyInfo is false.
+public class PFxExpressionYamlConverter : IYamlTypeConverter
+{
+    private readonly PFxExpressionYamlFormattingOptions _formattingOptions;
+
+    public PFxExpressionYamlConverter(PFxExpressionYamlFormattingOptions formattingOptions)
+    {
+        _formattingOptions = formattingOptions ?? throw new ArgumentNullException(nameof(formattingOptions));
+    }
+
+    public bool Accepts(Type type)
+    {
+        return type == typeof(PFxExpressionYaml);
+    }
+
+    public object? ReadYaml(IParser parser, Type type)
+    {
+        if (parser.TryConsumeNull())
+            return null;
+
+        var scalar = parser.Consume<Scalar>();
+
+        // Detect canonical format, where expressions should always start with '=' character
+        if (scalar.Value.Length >= 1 && scalar.Value[0] == PFxExpressionYamlFormattingOptions.ScalarPrefix)
+        {
+            var script = scalar.Value[1..];
+            return new PFxExpressionYaml(script);
+        }
+
+        throw new PaYamlSerializationException($"Power Fx expressions must start with '{PFxExpressionYamlFormattingOptions.ScalarPrefix}'.", scalar.Start);
+    }
+
+    public void WriteYaml(IEmitter emitter, object? value, Type type)
+    {
+        var expression = (PFxExpressionYaml?)value;
+        if (expression is null)
+        {
+            emitter.EmitNull();
+            return;
+        }
+
+        // If the script contains a substring that would cause extra yaml escaping, force it to a literal block
+        // The goal here for PFx yaml is that we avoid using yaml string escaping whenever possible.
+        var forceLiteralBlock =
+            // These char sequences are special to YAML parsers, so must be escaped or forced to literal block
+            expression.InvariantScript.Contains(" #")
+            || expression.InvariantScript.Contains(": ")
+            // Force multi-line scripts to be literal blocks
+            || expression.InvariantScript.Contains('\n')
+            || expression.InvariantScript.Contains('\r');
```

Comment:
```suggestion
            // Force multi-line scripts to be literal blocks
            || expression.InvariantScript.Contains('\n')
            || expression.InvariantScript.Contains('\r')
            || expression.InvariantScript.Contains('\u0085')  // Unicode next line
            || expression.InvariantScript.Contains('\u2028')  // Unicode line separator
            || expression.InvariantScript.Contains('\u2029'); // Unicord paragraph separator
```

Reply:
turns out that's actually problematic.
Since the original code doesn't handle these special Unicode chars, I'm going to leave it out for future work.

Reply:
won't fix.

Reply:
> Since the original code doesn't handle these special Unicode chars

Incorrect, it was handled in [ControlPropertyConverter.cs](https://github.com/microsoft/PowerApps-Tooling/blob/6e96cccd53d30ec52a034b076fa414a3769d95c7/src/Persistence/Yaml/ControlPropertyConverter.cs#L13)